qualtran/_infra/composite_bloq.py

```diff
@@ -49,11 +50,13 @@
     from qualtran.resource_counting import BloqCountT, SympySymbolAllocator
     from qualtran.simulation.classical_sim import ClassicalValT
 
+# NDArrays must be bound to np.generic
+SoquetType = TypeVar('SoquetType', bound=np.generic)
```

Comment:
can we make this private, i.e. `_SoquetType`? or will it be showing up in more places?

Reply:
Done.